kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyReconciler.java

```diff
@@ -142,15 +147,11 @@ private static KafkaProxy buildStatus(KafkaProxy primary,
     private static List<io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.Clusters> clusterConditions(ZonedDateTime now,
                                                                                                              KafkaProxy primary,
                                                                                                              Context<KafkaProxy> context) {
-        if (primary.getSpec() == null
-                || primary.getSpec().getClusters() == null) {
-            return List.of();
-        }
-        return primary.getSpec().getClusters().stream().map(cluster -> {
+        return ResourcesUtil.clustersInNameOrder(context).map(cluster -> {
```

Comment:
Why do we care about name order? Ideally we'd stick to YAML order as thats what the users of the operator will see, but I'm not sure if the parsers or the YAML spec actually support preserving that order. I guess name order at least gives us a consistency.

Reply:
virtual clusters are getting pulled out of the context as a Set, and we want them to be configured in a deterministic order for port allocation, so that that virtual cluster A always gets port 1,2,3 and virtual cluster B always gets 4,5,6. (like everything else, we want this to change in future, the current port allocation strategy is risky)

I don't think there is a YAML order any more as the VirtualKafkaClusters are independently defined in distinct custom resources.